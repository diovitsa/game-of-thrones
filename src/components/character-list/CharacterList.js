import React, { useEffect } from 'react';
import { Text, View } from 'react-native';
import Character from '../character/Character';
import { fetchCharacters } from '../../actions/characters';
import { connect } from 'react-redux';
import LoadingBar from '../loading-bar/LoadingBar';
import { charactersSelector } from '../../selectors';

export function CharacterList({ house, characters, fetchCharacters }) {

  useEffect(() => {
    fetchCharacters(house);
  }, []);

  const charactersList = characters
    ? characters.data.map((character, index) => {
      return <Character key={index} character={character}/>
    })
    : null;

  return (
    <View>
      <Text>Characters: </Text>
      {charactersList || <LoadingBar/>}
    </View>
  );
}

const mapStateToProps = (state, ownProps) => ({
  characters: charactersSelector(state, ownProps),
});

const mapDispatchToProps = {
  fetchCharacters,
};

export default connect(mapStateToProps, mapDispatchToProps)(CharacterList);