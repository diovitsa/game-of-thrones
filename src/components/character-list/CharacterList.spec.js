import React from 'react';
import { CharacterList } from './CharacterList';
import renderer from 'react-test-renderer'

describe('CharacterList', () => {
  let component;
  // let componentInstance;

  const emptyCharactersProps = {
    house: 'testHouse',
    fetchCharacters: () => {
    },
    characters: null
  };

  const propsWithCharacters = {
    ...emptyCharactersProps,
    characters: { data: [{ name: 'testName1' }, { name: 'testName2' }] }
  };

  describe('when loading in progress', () => {
    it('should show loading bar', () => {
      component = renderer.create(<CharacterList {...emptyCharactersProps}/>).toJSON();
      expect(component).toMatchSnapshot()
    });
  });

  describe('when characters are loaded', () => {
    it('should show loading bar', () => {
      component = renderer.create(
        <CharacterList {...propsWithCharacters}/>).toJSON();
      expect(component).toMatchSnapshot()
    });
  });

});