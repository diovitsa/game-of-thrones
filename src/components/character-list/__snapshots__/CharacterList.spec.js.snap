// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterList when characters are loaded should show loading bar 1`] = `
<View>
  <Text>
    Characters: 
  </Text>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "blue",
          "fontSize": 14,
        }
      }
    >
      testName1
    </Text>
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "blue",
          "fontSize": 14,
        }
      }
    >
      testName2
    </Text>
  </View>
</View>
`;

exports[`CharacterList when loading in progress should show loading bar 1`] = `
<View>
  <Text>
    Characters: 
  </Text>
  <Text>
    Loading in progress...
  </Text>
</View>
`;
